src/main/groovy/io/franzbecker/gradle/lombok/LombokPlugin.groovy

```diff
@@ -58,12 +58,12 @@ class LombokPlugin implements Plugin<Project> {
             )
         }
 
-        project.configurations.getByName(JavaPlugin.COMPILE_ONLY_CONFIGURATION_NAME).extendsFrom(configuration)
-        project.configurations.getByName(JavaPlugin.TEST_COMPILE_ONLY_CONFIGURATION_NAME).extendsFrom(configuration)
         boolean atLeastGradle4_6 = GradleVersion.version(project.gradle.gradleVersion) >= GradleVersion.version('4.6')
         if (atLeastGradle4_6) {
+            project.configurations.getByName(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME).extendsFrom(configuration)
             project.configurations.getByName(JavaPlugin.ANNOTATION_PROCESSOR_CONFIGURATION_NAME).extendsFrom(configuration)
-            project.configurations.getByName(JavaPlugin.TEST_ANNOTATION_PROCESSOR_CONFIGURATION_NAME).extendsFrom(configuration)
+        } else {
+            project.configurations.getByName("compile" /* JavaPlugin.COMPILE_CONFIGURATION_NAME */).extendsFrom(configuration)
```

Comment:
Why add it to "compile" and not compileOnly?

Reply:
When I tested compileOnly + grade 4.5 in my IntelliJ, it doesn't work in test source set.

comfileOnly => doesn't work
comfileOnly + testCompileOnly => works
compile => works

I haven't tested the older versions so I thought that it would be safe to revert back to the original for <4.6.

Reply:
I think it was „lombok“ initially

Reply:
I don't know what you mean by „lombok“. What I meant was revert back to the version prior to apply the below PR.

https://github.com/franzbecker/gradle-lombok/pull/35/files#diff-8aa499200ee24f9094443f6e0d9e69f9L60

It was something like this.

```
def compile = project.configurations.getByName(JavaPlugin.COMPILE_CONFIGURATION_NAME)
compile.extendsFrom(configuration)
```

But the `JavaPlugin.COMPILE_CONFIGURATION_NAME` has tagged as deprecated, I just put the string value as constant.

Reply:
Ah okay I see... should work. Thanks for testing it out!